src/app/controllers/public-forms.server.controller.js

```diff
@@ -44,9 +44,19 @@ exports.isFormPublic = function (req, res, next) {
  * @param  {Object} res - Express response object
  */
 exports.redirect = async function (req, res) {
+  const queryPortion = req.url ? req.url.split('?', 2)[1] : undefined
   let redirectPath = req.params.state
     ? req.params.Id + '/' + req.params.state
     : req.params.Id
+  if (
+    queryPortion &&
+    queryPortion.length > 0 &&
+    queryPortion.match(
+      /^(([0-9A-Za-z]+=[0-9A-Za-z]+)(&[0-9A-Za-z]+=[0-9A-Za-z]+)*)?$/,
```

Comment:
since this is a redirect endpoint, perhaps we can simplify the abstraction and forward all query parameters without checks

Reply:
don't we need to support special characters as well? I think it's quite unexpected for the user if they type a comma or full stop and suddenly their prefill stops working

Reply:
got it, I've relaxed the validation for this